src/errors.js

```diff
@@ -1,8 +1,10 @@
 'use strict'
 
 export class UndefinedFactError extends Error {
-  constructor (...props) {
+  constructor (factId, ...props) {
+    if (!props[0]) props[0] = `Undefined fact: ${factId}`
     super(...props)
```

Comment:
Let's find a way to ensure error.message is still set to something human readable; we can call ```super(`Undefined fact: ${factId}`)``` here to achieve that.